actionmailer/lib/action_mailer/base.rb

```diff
@@ -900,15 +913,19 @@ def collect_responses(headers)
           yield(collector)
           collector.responses
         elsif headers[:body]
-          [{
-            body: headers.delete(:body),
-            content_type: self.class.default[:content_type] || "text/plain"
-          }]
+          collect_responses_from_text(headers)
         else
           collect_responses_from_templates(headers)
         end
       end
 
+      def collect_responses_from_text(headers)
+        [{
+          body: headers.delete(:body),
+          content_type: headers[:content_type] || self.class.default[:content_type] || "text/plain"
```

Comment:
Is there another case different of those two?

If `:content_type` is passed to `mail`, `headers[:content_type]` will be the value passed to mail. In this case no matter what `self.class.default[:content_type]` is.

In the case you show the conditional:

```
headers[:content_type] || self.class.default[:content_type] || "text/plain"
```

will return the value of `:content_type` passed to `mail`, that is the same as `headers[:content_type]`, unless I'm missing something.